src/os/KeyValueStore.cc

```diff
@@ -1981,13 +1981,56 @@ int KeyValueStore::_zero(coll_t cid, const ghobject_t& oid, uint64_t offset,
                          size_t len, BufferTransaction &t)
 {
   dout(15) << __func__ << " " << cid << "/" << oid << " " << offset << "~" << len << dendl;
+  int r;
+  StripObjectMap::StripObjectHeaderRef header;
 
-  bufferptr bp(len);
-  bp.zero();
-  bufferlist bl;
-  bl.push_back(bp);
-  int r = _write(cid, oid, offset, len, bl, t);
+  r = t.lookup_cached_header(cid, oid, &header, true);
```

Comment:
I don't think the DB updates will be less by this change. we can zero out the buffer and write some same sane data after some time. This change will involve removing the keys and writing them again. Not sure how frequent we can get _zero().